ibis/sql/mssql/api.py

```diff
@@ -0,0 +1,125 @@
+from ibis.sql.alchemy import to_sqlalchemy
+from ibis.sql.mssql.client import MSSQLClient
+from ibis.sql.mssql.compiler import (  # noqa: F401, E501
+    compiles,
+    dialect,
+    rewrites,
+)
+
+
+def compile(expr, params=None):
+    """Compile an ibis expression to the MSSQL target.
+
+    Parameters
+    ----------
+    expr : ibis.expr.types.Expr
+        The ibis expression to compile
+    params : dict or None
+        ``dict`` mapping :class:`ibis.expr.types.ScalarParameter` objects to
+        values
+
+    Returns
+    -------
+    sqlalchemy_expression : sqlalchemy.sql.expression.ClauseElement
+
+    Examples
+    --------
+    >>> import os
+    >>> import getpass
+    >>> host = os.environ.get('IBIS_TEST_MSSQL_HOST', 'localhost')
+    >>> user = os.environ.get('IBIS_TEST_MSSQL_USER', getpass.getuser())
+    >>> password = os.environ.get('IBIS_TEST_MSSQL_PASSWORD')
+    >>> database = os.environ.get('IBIS_TEST_MSSQL_DATABASE',
+    ...                           'ibis_testing')
+    >>> con = connect(
+    ...     database=database,
+    ...     host=host,
+    ...     user=user,
+    ...     password=password
+    ... )
+    >>> t = con.table('functional_alltypes')
+    >>> expr = t.double_col + 1
+    >>> sqla = compile(expr)
+    >>> print(str(sqla))  # doctest: +NORMALIZE_WHITESPACE
+    SELECT t0.double_col + %(param_1)s AS tmp
+    FROM functional_alltypes AS t0
+    """
+    return to_sqlalchemy(expr, dialect.make_context(params=params))
+
+
+def connect(
+    host='localhost',
+    user=None,
+    password=None,
+    port=1433,
+    database=None,
+    driver='pyodbc',
+    odbc_driver=None,
+    url=None,
+):
+    """Create an Ibis client located at `user`:`password`@`host`:`port`
+    connected to a MSSQL database named `database`.
+
+    Parameters
+    ----------
+    host : string, default 'localhost'
+    user : string, default None
```

Comment:
instead of `default None`, change it to `optional`. eg: `user : string, optional`
the same for the other parameters.

Reply:
Fixed.